django/thunderstore/repository/models/comment.py

```diff
@@ -0,0 +1,44 @@
+from django.conf import settings
+from django.contrib.contenttypes.fields import GenericForeignKey
+from django.contrib.contenttypes.models import ContentType
+from django.db import models
+from ulid2 import generate_ulid_as_uuid
+
+from thunderstore.core.mixins import TimestampMixin
+
+
+class Comment(TimestampMixin, models.Model):
+    thread = GenericForeignKey("thread_content_type", "thread_object_id")
+    thread_content_type = models.ForeignKey(
+        ContentType,
+        on_delete=models.CASCADE,
+        related_name="comments",
+    )
+    thread_object_id = models.PositiveIntegerField()
```

Comment:
A problem: Some of our models use UUID for their primary key, some use integer

Reply:
```suggestion
    thread_object_id = models.CharField(max_length=36)
```
This works, do you have something else in mind?

Reply:
Does this work with both integer and UUID based?

Additionally, since an UUID is a 128-bit number, we might be able to use an UUID field (assuming incremental numbers are compatible). Could also see if Django has a 128-bit number field support via some avenue (how is UUID field defined behind the scenes exactly?)

Reply:
> Does this work with both integer and UUID based?

Yes, I tested it and it worked fine (and made sure it fails with a field that shouldn't work)

Reply:
> we might be able to use an UUID field

Yes, this works too. Do you prefer using a `UUIDField` over `CharField` here?

Reply:
If it works then yes. The postgres docs mention that the UUID datatype always returns data in the uuid format, which makes me wonder if it can lead to issues with sequential private keys though

Reply:
Not sure but it definitely stores correctly. There are some docs about using a [different field](https://docs.djangoproject.com/en/3.1/ref/contrib/contenttypes/#django.contrib.contenttypes.fields.GenericForeignKey.for_concrete_model) to the original primary key.